pkg/client/cache/fifo.go

```diff
@@ -39,6 +44,16 @@ type Queue interface {
 	HasSynced() bool
 }
 
+// Helper function for popping from Queue.
+func Pop(queue Queue) interface{} {
```

Comment:
@lavalamp - since this is already merged and we probably need to fix some more issues - see https://github.com/kubernetes/kubernetes/issues/27004#issuecomment-225963877 (but those one are very local changes) I will update the comment in that PR tomorrow.